tests/Fixer/Whitespace/BlankLineBeforeControlStatementFixerTest.php

```diff
@@ -0,0 +1,1046 @@
+    {
+        $this->fixer->configure(array(
+            'continue',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithContinue()
+    {
+        return array(
+            array(
+'<?php
+while (true) {
+    continue;
+}',
+            ),
+            array(
+'<?php
+while (true) {
+    continue 1;
+}',
+            ),
+            array(
+'<?php
+while (true) {
+    while (true) {
+        continue 2;
+    }
+}',
+            ),
+            array(
+'<?php
+while (true) {
+    $foo = true;
+
+    continue;
+}',
+'<?php
+while (true) {
+    $foo = true;
+    continue;
+}',
+            ),
+            array(
+'<?php
+while (true) {
+    $foo = true;
+
+    continue 1;
+}',
+'<?php
+while (true) {
+    $foo = true;
+    continue 1;
+}',
+            ),
+            array(
+'<?php
+while (true) {
+    while (true) {
+        $foo = true;
+
+        continue 2;
+    }
+}',
+'<?php
+while (true) {
+    while (true) {
+        $foo = true;
+        continue 2;
+    }
+}',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithDeclare
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithDeclare($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'declare',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithDeclare()
+    {
+        return array(
+            array(
+'<?php
+declare(ticks=1);',
+            ),
+            array(
+'<?php
+
+$foo = "bar";
+
+declare(ticks=1);',
+'<?php
+
+$foo = "bar";
+declare(ticks=1);',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithDo
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithDo($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'do',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithDo()
+    {
+        return array(
+            array(
+'<?php
+do {
+} while (true);',
+            ),
+            array(
+'<?php
+$foo = "bar";
+
+do {
+} while (true);',
+'<?php
+$foo = "bar";
+do {
+} while (true);',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithElse
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithElse($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'else',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithElse()
+    {
+        return array(
+            array(
+'<?php
+if (true) {
+    echo $b;
+} 
+
+else {
```

Comment:
So this is what passes the tests, but personally, I would want it to fix this to

```php
<?php
if (true) {
    echo $b;

} else {
    echo $c;
}
```

Well, actually, I wouldn't want it to be fixed to that, but I _think_ that maybe that is what other would want this to be fixed to, what do you think?